warehouse/validations/validations.go

```diff
@@ -50,7 +51,8 @@ func Init() {
 	connectionTestingFolder = config.GetString("RUDDER_CONNECTION_TESTING_BUCKET_FOLDER_NAME", misc.RudderTestPayload)
 	pkgLogger = logger.NewLogger().Child("warehouse").Child("validations")
 	fileManagerFactory = filemanager.New
-	objectStorageValidationTimeout = 15 * time.Second
+	objectStorageTimeout = config.GetDuration("Warehouse.Validations.ObjectStorageTimeout", 15, time.Second)
+	queryTimeout = config.GetDuration("Warehouse.Validations.QueryTimeout", 25, time.Second)
```

Comment:
Increasing the default timeout from 15 seconds to 25 seconds as sometimes it takes more than 15 seconds. Also, we need to keep it below cp router default timeout (30 seconds).

Reply:
The fact that we need to keep it below 30 seconds and why is good info, we could have a comment about it here right before `queryTimeout = config.GetDuration(...)`.

Reply:
Sure.

Reply:
Making changes in [here](https://github.com/rudderlabs/rudder-server/pull/3883).